db/table_cache.cc

```diff
@@ -224,7 +224,8 @@ InternalIterator* TableCache::NewIterator(
   }
   InternalIterator* result = nullptr;
   if (s.ok()) {
-    result = table_reader->NewIterator(options, arena, skip_filters);
+    result =
+      table_reader->NewIterator(options, arena, icomparator, skip_filters);
```

Comment:
Does comparator need to be passed here? It was already passed to `NewTableReader()` in `TableReaderOptions`. It seems more straightforward to propagate it from there, for table types that need it. In particular, for block-based tables, it's already propagated: BlockBasedTable::Rep::internal_comparator.